### ISSUE TITLE
Hi, I want to train my LED, how many pictures do I need?

### ISSUE THREAD

**Issue by yangxiaoming230** [QtacierP/LED#2]
Thank you very much for your work. I recently tried to train my own LED, following the instructions in your Readme. But my test results weren't very good. I would like to ask how many images you used in your data set, are they all the high-quality images in the EyeQ data set? Are there any other details to keep in mind when training?

**Comment by QtacierP** [comment]
Yes. We use all high quality images in EyeQ train dataset. Maybe you should enlarge the number of epochs if your dataset is relatively small.

**Comment by QtacierP** [comment]
Also, the degradation model/paird data is important. If you find tje model performs well on the training set but fails on the real data, try a new degradation framework please.

One way is that you can use t-SNE to visualize and judge the distribution gap between the degraded images and realistic low quality images, like the figure 1 in the paper.

**Comment by yangxiaoming230** [comment]
Thanks for your reply. I only used 600 images to train LED last time. I estimate that EyeQ has about 16,000 high quality images. But my lab hardware doesn't have enough arithmetic power to train such a large number of images. Is there any other way to overcome this problem?

**Comment by QtacierP** [comment]
Do you also train on the fundus images (or other similar modalities)? If so, maybe you could fine-tune the pre-trained weights we have provided. In case computational resources are limited, employing a PEFT method such as LoRA might prove beneficial. Otherwise, training a diffusion model from scratch with a limited dataset would pose a challenge. Our future work primarily focuses on developing a diffusion model for image restoration, incorporating prior regularization. This approach is expected to yield better results with smaller datasets. Once we publish our paper and release the codes, we will notify you accordingly.

**Comment by yangxiaoming230** [comment]
Yes, I use fundus images EyeQ as well. But I only use part of it. I will learn the methods you suggested. Sincerely thanks for your help. I will continue to follow your research as well.

**Comment by askerlee** [comment]
hi @QtacierP thanks for sharing this good model. Seems the paper was not accepted to MICCAI'23? It's quite a regret.

**Comment by QtacierP** [comment]
@askerlee Thank you for your attention. The current version of LED has not been accepted yet. It still has several shortcomings, particularly in relation to classical reference metrics such as PSNR and SSIM. We are currently developing LED-V2 to achieve better performance and improved generalization. We hope to release it as soon as possible.

**Comment by askerlee** [comment]
Great to learn the progress. May I know if the shortcomings are due to randomness of diffusion? I ran denoising on the same image multiple times and noticed the resulted images are slightly different from each other. But this is an inherent issue of diffusion and hard to solve, IMHO.

**Comment by QtacierP** [comment]
Thank you for your kind sharing. Indeed, Diffusion Models are powerful generative models due to their variability. However, medical image translation (such as restoration and modality transformation) is quite different from AIGC. In these cases, ODE-based Diffusion Models or one-step models (like Consistency Model) might be more suitable.

**Comment by askerlee** [comment]
Hmmm I see. These variants are of smaller variances and may recover the groundtruth better. But please don't lose heart in diffusion models! Maybe you need some auxiliary information to help reduce the "search space" of the vanilla diffusion models, e.g. something like Image prompts (similar idea as IP-Adapter but for medical images).

**Comment by QtacierP** [comment]
Thank you for your kind and helpful suggestions! Prompts indeed seem to be a promising solution for controlling the direction in the diffusion recovery process. I will continue this work and strive to generalize it beyond fundus imaging in the medical domain. Once again, thank you for your valuable input :)
